### PR TITLE
Parameter support for DBI dbGetQuery

### PR DESCRIPTION
**Pull request by byapparov**
Closes: #444

This would help to implement something similar to [BigQuery Magics](https://googleapis.dev/python/bigquery/latest/magics.html) in Rmarkdown https://github.com/yihui/knitr/issues/1867

### CODE REVIEW

**Comment by byapparov** [comment]
@hadley this is ready for review.

File: R/bq-perform.R [hunk 1]
```diff
@@ -263,7 +263,7 @@ bq_perform_query <- function(query, billing,
     priority = unbox(priority)
   )
 
-  if (!is.null(parameters)) {
+  if (!is.null(parameters) && length(parameters) > 0) {
```
Comment (hadley) [inline R/bq-perform.R:266]:
```suggestion
  if (length(parameters) > 0) {
```

File: R/dbi-result.R [hunk 2]
```diff
@@ -2,10 +2,21 @@
 NULL
 
 BigQueryResult <- function(conn, sql, ...) {
+
+  # extract params of the query if provided
+  # to support DBI interface for parameterized queries
+  args <- c(as.list(environment()), list(...))
```
Comment (hadley) [inline R/dbi-result.R:8]: Why not add `params` to the argument list?

**Comment by byapparov** [comment]
@hadley I have updated the changes according to your comments, thanks

**Comment by hadley** [comment]
Closing in favour of #573

**Pull request closed without merging**